src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -220,6 +220,21 @@ ApplicationSharedPtr FindApp(DataAccessor<ApplicationSet> accessor,
 }
 
 template <class UnaryPredicate>
+ApplicationSharedPtr FindApp(
+    DataAccessor<ApplicationManagerImpl::AppsWaitRegistrationSet> accessor,
+    UnaryPredicate finder) {
+  ApplicationManagerImpl::AppsWaitRegistrationSet::iterator it = std::find_if(
```

Comment:
@AGaliuzov, perhaps you right. But the same way released search applications in `ApplicationManagerImpl::applications_`. So, I think that better use the same solution for all lists. In this way it easier support code than every time new solution.